### PR TITLE
CTCTRADERS-2411: add Ni offices behind toggle

### PR DESCRIPTION
**Pull request by kenneth-rayner**
…bled

### CODE REVIEW

**Comment by kenneth-rayner** [comment]
Test this please

File: app/connectors/ReferenceDataConnector.scala [hunk 1]
```diff
@@ -35,12 +36,9 @@ class ReferenceDataConnector @Inject()(config: FrontendAppConfig, http: HttpClie
     }
   }
 
-  def getCustomsOfficesOfTheCountry(country: String)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Seq[CustomsOffice]] = {
-    val serviceUrl = s"${config.referenceDataUrl}/customs-offices/$country"
-
-    metricsService.timeAsyncCall(Monitors.getCustomsOfficesOfTheCountryMonitor) {
```
Comment (ericohwotu) [inline app/connectors/ReferenceDataConnector.scala:41]: any reason why this was removed? curious?

File: test/controllers/SimplifiedCustomsOfficeControllerSpec.scala [hunk 2]
```diff
@@ -180,10 +187,9 @@ class SimplifiedCustomsOfficeControllerSpec extends SpecBase with AppWithDefault
       Json.obj("value" -> "officeId", "text" -> "someName (officeId)", "selected" -> false)
     )
 
-    when(mockRenderer.render(any(), any())(any()))
-      .thenReturn(Future.successful(Html("")))
-
-    when(mockRefDataConnector.getCustomsOfficesOfTheCountry(Matchers.eq("GB"))(any(), any())).thenReturn(Future.successful(customsOffices))
+    when(mockRenderer.render(any(), any())(any())).thenReturn(Future.successful(Html("")))
+    when(mockRefDataConnector.getCustomsOfficesOfTheCountry(any())(any(), any())).thenReturn(Future.successful(customsOffices))
```
Comment (ericohwotu) [inline test/controllers/SimplifiedCustomsOfficeControllerSpec.scala:191]: do you need this and the next mock?

File: test/controllers/SimplifiedCustomsOfficeControllerSpec.scala [hunk 3]
```diff
@@ -217,8 +223,9 @@ class SimplifiedCustomsOfficeControllerSpec extends SpecBase with AppWithDefault
   private def verifyStatusAndContent(customsOfficeJson: Seq[JsObject], boundForm: Form[CustomsOffice], result: Future[Result], expectedStatus: Int): Any = {
     status(result) mustEqual expectedStatus
 
+    when(mockRefDataConnector.getCustomsOfficesOfTheCountry(anyObject())(any(), any())).thenReturn(Future.successful(customsOffices))
```
Comment (ericohwotu) [inline test/controllers/SimplifiedCustomsOfficeControllerSpec.scala:226]: Same here

File: app/services/CustomsOfficesService.scala [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+)(implicit ec: ExecutionContext) {
+
+  private def getNICustomsOffices(implicit hc: HeaderCarrier): Future[CustomsOfficeList] =
+    if (frontendAppConfig.isNIJourneyEnabled) {
+      referenceDataConnector.getCustomsOfficesOfTheCountry(CountryCode("XI"))
+    } else {
+      Future.successful(CustomsOfficeList(Nil))
+    }
+
+  def getCustomsOfficesOfArrival(implicit hc: HeaderCarrier): Future[CustomsOfficeList] =
```
Comment (ericohwotu) [inline app/services/CustomsOfficesService.scala:40]: test for this function please

File: app/connectors/ReferenceDataConnector.scala (hunk 1 above)
Comment (kenneth-rayner) [reply app/connectors/ReferenceDataConnector.scala:41]: Refactored it into a new function to use the model rather than a string

Comment (ericohwotu) [reply app/connectors/ReferenceDataConnector.scala:41]: My bad, was referring to the metrics call

File: test/controllers/CustomsOfficeControllerSpec.scala [hunk 5]
```diff
@@ -85,10 +89,9 @@ class CustomsOfficeControllerSpec extends SpecBase with AppWithDefaultMockFixtur
 
     "must redirect to session expired page when user hasn't answered the customs sub place question" in {
 
-      when(mockRenderer.render(any(), any())(any()))
-        .thenReturn(Future.successful(Html("")))
-
-      when(mockRefDataConnector.getCustomsOfficesOfTheCountry(anyObject())(any(), any())).thenReturn(Future.successful(customsOffices))
+      when(mockRenderer.render(any(), any())(any())).thenReturn(Future.successful(Html("")))
+      when(mockRefDataConnector.getCustomsOfficesOfTheCountry(any())(any(), any())).thenReturn(Future.successful(customsOffices))
```
Comment (ericohwotu) [inline test/controllers/CustomsOfficeControllerSpec.scala:93]: Connector call

File: test/controllers/CustomsOfficeControllerSpec.scala [hunk 6]
```diff
@@ -102,8 +105,9 @@ class CustomsOfficeControllerSpec extends SpecBase with AppWithDefaultMockFixtur
     }
 
     "must redirect to the next page when valid data is submitted" in {
-      when(mockRefDataConnector.getCustomsOfficesOfTheCountry(anyObject())(any(), any())).thenReturn(Future.successful(customsOffices))
+      when(mockRefDataConnector.getCustomsOfficesOfTheCountry(any())(any(), any())).thenReturn(Future.successful(customsOffices))
```
Comment (ericohwotu) [inline test/controllers/CustomsOfficeControllerSpec.scala:108]: Connector call

File: test/controllers/CustomsOfficeControllerSpec.scala [hunk 7]
```diff
@@ -34,32 +34,36 @@ import play.api.mvc.Result
...
   override def beforeEach(): Unit = {
     super.beforeEach()
     Mockito.reset(mockRefDataConnector)
+    Mockito.reset(mockCustomsOfficesService)
   }
 
   override def guiceApplicationBuilder(): GuiceApplicationBuilder =
     super
       .guiceApplicationBuilder()
       .overrides(bind[ReferenceDataConnector].toInstance(mockRefDataConnector))
```
Comment (ericohwotu) [inline test/controllers/CustomsOfficeControllerSpec.scala:65]: This can be rmoved

File: test/controllers/CustomsOfficeControllerSpec.scala [hunk 8]
```diff
@@ -178,10 +183,9 @@ class CustomsOfficeControllerSpec extends SpecBase with AppWithDefaultMockFixtur
       Json.obj("value" -> "officeId", "text" -> "someName (officeId)", "selected" -> false)
     )
 
-    when(mockRenderer.render(any(), any())(any()))
-      .thenReturn(Future.successful(Html("")))
-
-    when(mockRefDataConnector.getCustomsOfficesOfTheCountry(anyString())(any(), any())).thenReturn(Future.successful(customsOffices))
+    when(mockRenderer.render(any(), any())(any())).thenReturn(Future.successful(Html("")))
+    when(mockRefDataConnector.getCustomsOfficesOfTheCountry(any())(any(), any())).thenReturn(Future.successful(customsOffices))
```
Comment (ericohwotu) [inline test/controllers/CustomsOfficeControllerSpec.scala:187]: same

File: test/controllers/SimplifiedCustomsOfficeControllerSpec.scala [hunk 9]
```diff
@@ -88,7 +92,8 @@ class SimplifiedCustomsOfficeControllerSpec extends SpecBase with AppWithDefault
       when(mockRenderer.render(any(), any())(any()))
         .thenReturn(Future.successful(Html("")))
 
-      when(mockRefDataConnector.getCustomsOfficesOfTheCountry(anyString())(any(), any())).thenReturn(Future.successful(customsOffices))
+      when(mockRefDataConnector.getCustomsOfficesOfTheCountry(any())(any(), any())).thenReturn(Future.successful(customsOffices))
```
Comment (ericohwotu) [inline test/controllers/SimplifiedCustomsOfficeControllerSpec.scala:95]: connector

**Review by ericohwotu — APPROVED** [review]

**Pull request merged**